tests/comments/test_async_serializer.py

```diff
@@ -0,0 +1,229 @@
+
+    # Testing a comment that has been censored by a moderator:
+    comment_4 = comment_data[3]
+    assert comment_4['id'] == 4
+    assert comment_4['user_name'] == 'unknown user'
+    assert comment_4['user_pk'] == -1
+    assert comment_4['user_profile_url'] == ''
+    assert comment_4['is_deleted'] is True
+    assert comment_4['is_moderator'] is False
+    assert len(comment_4['child_comments']) == 0
+    assert comment_4['comment'] == Comment.objects.get(pk=4).comment
+    assert comment_4['comment'] == ''
+    assert comment_4['object_pk'] == question.pk
+    assert comment_4['is_removed'] is False
+    assert comment_4['is_censored'] is True
+    assert comment_4['is_blocked'] is False
+    assert len(comment_4['comment_categories']) == 0
+    assert comment_4['last_discussed'] is None
+    assert comment_4['is_moderator_marked'] is False
+    assert comment_4['content_type'] == question_ct.pk
+
+    # Testing a comment that has been deleted by its creator:
+    comment_5 = comment_data[2]
+    assert comment_5['id'] == 5
+    assert comment_5['user_name'] == 'unknown user'
+    assert comment_5['user_pk'] == -1
+    assert comment_5['user_profile_url'] == ''
+    assert comment_5['is_deleted'] is True
+    assert comment_5['is_moderator'] is False
+    assert len(comment_5['child_comments']) == 0
+    assert comment_5['comment'] == Comment.objects.get(pk=5).comment
+    assert comment_5['comment'] == ''
+    assert comment_5['object_pk'] == question.pk
+    assert comment_5['is_removed'] is True
+    assert comment_5['is_censored'] is False
+    assert comment_5['is_blocked'] is False
+    assert len(comment_5['comment_categories']) == 0
+    assert comment_5['last_discussed'] is None
+    assert comment_5['is_moderator_marked'] is False
+    assert comment_5['content_type'] == question_ct.pk
+
+    # What is the deleted status of comments that have been
+    # blocked, censored, and removed?
+    # A blocked comment should not be deleted in case it must be unblocked:
+    assert comment_3['is_blocked'] is True
+    assert comment_3['is_deleted'] is False
+
+    # A censored comment is actually is_deleted.
+    assert comment_4['is_censored'] is True
+    assert comment_4['is_deleted'] is True
+    # But does this make sense or should the content remain accessible?
+
+    # A removed comment should be is_deleted:
+    assert comment_5['is_removed'] is True
+    assert comment_5['is_deleted'] is True
+
+    # The difference between is_removed and is_censored,
+    # is that is_removed is carried out by the creator,
+    # and is_censored can only be done by a moderator:
+    assert comment_5['is_moderator'] is False
+    # But this differentiation does not occur in the serializer:
+    assert comment_4['is_moderator'] is False
+    # Because the serializer does not differentiate between
+    # moderator and creator for deleting the comment,
+    # does it actually make sense to show this
+    # differentiation in the serializer?
```

Comment:
The is_moderator field is True for comments added by a moderator. (As you can see in the comment added by a moderator in the test below). We use it to display a "moderator" badge with theses comments, where the creator was also moderator of the project.